Support YQL, Remove Webservice because it is no longer supported by Yahoo

mchaver opened the pull request.

mchaver committed:
Working functions, types and API for YQL

mchaver committed:
Start reorganizing the code

mchaver committed:
Split into internal files

mchaver committed:
servant-0.4 is compiled

mchaver committed:
Compiles with servant 0.4-0.9

mchaver committed:
Remove webserver api (no longer provided by Yahoo)

mchaver committed:
API test passing for all versions of servant

mchaver committed:
Parsers null returns for non-existent StockSymbols as Nothing

mchaver committed:
Clean up documentation and tests

mchaver committed:
Merge pull request #1 from mchaver/yql

Update package to query yql, remove deprecated webservice

cdepillabout committed:
Remove whitespace on ends of lines in README.md

cdepillabout committed:
Add additional shields and show required imports in example.

cdepillabout committed:
Fix some whitespace on ends of lines and rearrange some documentation.

cdepillabout committed:
Rename Web.Yahoo.Finance.YQL.Types module to Web.Yahoo.Finance.YQL.

cdepillabout committed:
Remove whitespaces at end of lines in FinanceSpec.

cdepillabout commented:
@mchaver Thanks a lot!

I've added you as a collaborator for this project.  Feel free to merge this PR in.  Also, if you tell me your hackage username I'll add you as an admin for the package on hackage.  That way you'll be able to do releases to hackage.

In 3984d93 I renamed the `Web.Yahoo.Finance.YQL.Types` module to `Web.Yahoo.Finance.YQL`.  I think this makes more sense, given that the module is also exporting the functions for actually getting stock quotes.  What doyou think?

Also, it appears that the tests are [failing on travis-ci](https://travis-ci.org/cdepillabout/yahoo-finance-api/builds/166312140).  I'm not sure why this is.  Do you have any ideas?  If we can't figure it out, I guess we can just disable the tests on travis-ci.

mchaver commented:
@cdepillabout 
Yeah, the rename makes sense.

My guess would be the network settings on the travis system that runs the tests. On travis the tests only pass on `#GHC 7.1.3`. I can get the tests `stack test` to pass on mac and ubuntu. I think it is fine to disable the travis tests and just make sure stack test works before pushing to hackage.

cdepillabout committed:
Don't run test in travis.

cdepillabout committed:
Update .travis.yml file.

Update the .travis.yml file with the recommended file from
https://docs.haskellstack.org/en/stable/GUIDE/#travis-with-caching.

Also enable additional travis targets (older versions of ghc).

cdepillabout committed:
Disable 7.6.3 and stack-default builds in .travis.yml.

cdepillabout committed:
Create stack-servant-0.9.yaml file.

This commit creates a stack-servant-0.9.yaml file.  This is used by the
test-stack-files.sh script to test that yahoo-finance-api works with
servant-0.9.

This commit also sets stack.yaml to just be a normal build using lts-7.

cdepillabout committed:
Import Control.Applicative when using old version of base.

cdepillabout committed:
Disable tests in .travis.yml when building with cabal.

cdepillabout committed:
Disable lts-2 builds in .travis.yml because servant version is too old.

cdepillabout commented:
I released v0.2.0.0 to hackage.  Thanks again!

Closes #3.
